### ISSUE TITLE
Why are there no real downstream evaluation?

### ISSUE THREAD

**Issue by RicherMans** [LZH-0225/AudioFormer#3]
Hey there,
I came across this repo and the paper and the results seem impressive.
However, it appears to me that there is no actual downstream evaluation done here since Encodec is specifically trained on Audioset + FSD50k.
Your paper somewhat boldly states that "on other downstream tasks, the method also works", but there is no real evidence that this is the case.
Thus I might ask why haven't you run say ESC-50? It's a pretty small dataset that wouldn't take much time or effort. 
If you have run these experiments, how were the results?
Kind regards,
Heinrich

**Comment by RicherMans** [comment]
Hey there,
thanks for the quick response!
> Of course, we accepted your suggestion and verified it on the ESC-50, but the effect was not satisfactory, with an accuracy rate of 87%.
So this is the AS-Finetuned model we are talking here right?

> We believe that this is not only influenced by EnCodec

I strongly disagree here, it seems obvious from my point of view. 
Usually improvement on Audioset in terms of mAP comes with more or less equal improvements on ESC-50, FSD50k and some DCASE tasks, since they are closely related.
As an example, your results outperform PaSST by a large margin ~ 5 point mAP on Audioset, however fail to obtain any reasonable improvement for FSD50k, which is a bit suspicious.
Just for reference, PaSSTs mAP is about 5 points higher than the one of PSLA (on Audioset), and obtained a 10 point mAP improvement on FSD50k.

Further, similar to your work, in audio we had similar trends of using raw-waveform with neural network extractors in the past, stuff like SincNet.
Directly learning features has always been a big goal in audio, such that we can finally also train models `end-to-end` like in vision and nlp.
Raw-wave generally obtains good results on in-domain datasets, I.e., datasets it has been trained on, but performance heavily degrades on out-domain datasets.
Thus we generally use mel-spectrograms, since these features are not "learned" and thus are not prone to "over/underfitting".


> Viewing audio tasks as natural language tasks also faces the problem of requiring a large amount of data for natural language tasks. 

Yeah maybe, but the problem is not the amount of data, the problem in audio is noise. Google trained speech recognition models on petabytes of data, but couldn't produce a  "one fits it all" speech recognition model.
Simple things like changing a microphone when recording a signal can lead to large performance degredations in audio. It's unreasonable to think that one would one day have data that encompasses all sorts of sounds with all sorts of microphones having all sorts of noise.

**Comment by LZH-0225** [comment]
The inputs of PassT and PSLA are both Mel spectrograms, and in my opinion, they essentially treat audio tasks as image tasks. It seems that image tasks can be well transferred, and most models based on Mel spectrum use ImageNet for pre training or use the pre training parameters of the image model. I don't know much about ImageNet, it seems that this dataset doesn't have Mel spectrum data. Why can I migrate to an audio task? I don't quite understand this point.
AudioFormer regards audio tasks as natural language comprehension tasks, but using pre trained parameters from text models can result in poor performance, which is intuitive. Not only for publicly available data, but also for other tasks, I use pre trained image models on ImageNet that can be well transferred to Mel spectrograms as input for audio tasks.

**Comment by RicherMans** [comment]
Hey there,
 > The inputs of PassT and PSLA are both Mel spectrograms, and in my opinion, they essentially treat audio tasks as image tasks. 
Yeah spectrograms have two dimensions, but thats the most similarity they have with audio. Images have spatial relationships and generally signals do not overlap (like a mouse behind a car is just not visible).
For audio, this is not the case, objects of interest commonly overlap (somebody speaks, in the background there are other people speaking, with also some sound of wind).

> It seems that image tasks can be well transferred, and most models based on Mel spectrum use ImageNet for pre training or use the pre training parameters of the image model. 
Yeah thats only true for classification, if it would be the case that Mel-spectrograms = Images and they can be pretrained using ImageNet, why did nobody in the field of automatic speech recognition use that? 

> I don't know much about ImageNet, it seems that this dataset doesn't have Mel spectrum data. 
Sure it just has images which are different from spectrograms, but are also signals, as audio.

> AudioFormer regards audio tasks as natural language comprehension tasks, but using pre trained parameters from text models can result in poor performance, which is intuitive. Not only for publicly available data, but also for other tasks, I use pre trained image models on ImageNet that can be well transferred to Mel spectrograms as input for audio tasks. 
So don't get me wrong here, I was hoping that your work would be a new stage of audio classification. 
Like finally we can get rid of some signal processing that is left in our pipelines.
My initial question was actually "does your performance transfer to other tasks?" and your response is essentially "no, only to data that EnCodec has seen".

While it's great to get good results on Audioset, your method seems not very useful in practice, since I would need to retrain EnCodec for each new sample.

Since it seems you haven't done audio-related work before, just a quick summary of why we use spectrograms:
1. Spectrograms are a linear transform between the input and the time-frequency representation, i.e., "image" whereas a spectrogram "image" contains distinct frequency information stored in different positions which are different from visual images. 
2.  Spectrograms are not trained and due to their linear nature, produce a predictable outcome with different microphone inputs.

For audio, any signal that you might encounter in the real world can be recorded by many different microphones. Many microphones may produce a very different signal, which is actually common.
The problem with learnable front-ends is that they generally rely on the input data being "seen" during training.

Thanks for the reply!
Kind regards,
Heinrich
